File: backend/Models/Ilan.cs
namespace Backend.Models;

public class Ilan
{
    public int Id { get; set; }
    public string Baslik { get; set; } = string.Empty;
    public string Aciklama { get; set; } = string.Empty;
    public string Meslek { get; set; } = string.Empty;
    public string Konum { get; set; } = string.Empty;
    public DateTime SonBasvuruTarihi { get; set; }
    public DateTime OlusturmaTarihi { get; set; } = DateTime.Now;

    public int IsverenId { get; set; }
    public Kullanici? Isveren { get; set; }

    public List<Basvuru> Basvurular { get; set; } = new();
}

File: mobile/lib/models/ilan.dart
class Ilan {
  final int id;
  final String baslik;
  final String aciklama;
  final String meslek;
  final String konum;
  final DateTime sonBasvuruTarihi;
  final int isverenId;
  final DateTime olusturmaTarihi;

  Ilan({
    required this.id,
    required this.baslik,
    required this.aciklama,
    required this.meslek,
    required this.konum,
    required this.sonBasvuruTarihi,
    required this.isverenId,
    required this.olusturmaTarihi,
  });

  factory Ilan.fromJson(Map<String, dynamic> json) {
    return Ilan(
      id: json['id'],
      baslik: json['baslik'],
      aciklama: json['aciklama'],
      meslek: json['meslek'],
      konum: json['konum'],
      sonBasvuruTarihi: DateTime.parse(json['sonBasvuruTarihi']),
      isverenId: json['isverenId'] ?? 0,
      olusturmaTarihi: json['olusturmaTarihi'] != null
          ? DateTime.parse(json['olusturmaTarihi'])
          : DateTime.now(),
    );
  }
}